### PR TITLE
More warning fixes

### PR DESCRIPTION
**Pull request by bwbarrett**
.

### CODE REVIEW

File: ompi/communicator/ft/comm_ft.c [hunk 1]
```diff
@@ -132,15 +134,21 @@ int ompi_comm_shrink_internal(ompi_communicator_t* comm, ompi_communicator_t** n
...
+#endif
     opal_mutex_lock(&ompi_group_afp_mutex);
     ompi_group_intersection(comm->c_remote_group, ompi_group_all_failed_procs, &failed_group);
     opal_mutex_unlock(&ompi_group_afp_mutex);
+#if OPAL_ENABLE_DEBUG
     stop = PMPI_Wtime();
+#endif
     OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
                          "%s ompi: comm_shrink: group_inter: %g seconds",
                          OMPI_NAME_PRINT(OMPI_PROC_MY_NAME), stop-start));
```
Comment (gpaulsen) [inline ompi/communicator/ft/comm_ft.c:148]: I think we also need `#if OPAL_ENABLE_DEBUG` guards around the `OPAL_OUTPUT_VERBSE()` call which prints `stop-start`?

File: ompi/communicator/ft/comm_ft.c [hunk 2]
```diff
@@ -154,7 +162,9 @@ int ompi_comm_shrink_internal(ompi_communicator_t* comm, ompi_communicator_t** n
                                         comm,
                                         comm->c_coll->coll_agree_module);
     } while( MPI_ERR_PROC_FAILED == ret );
+#if OPAL_ENABLE_DEBUG
     stop = PMPI_Wtime();
+#endif
     OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
                          "%s ompi: comm_shrink: AGREE: %g seconds",
                          OMPI_NAME_PRINT(OMPI_PROC_MY_NAME), stop-start));
```
Comment (gpaulsen) [inline ompi/communicator/ft/comm_ft.c:170]: I think we also need `#if OPAL_ENABLE_DEBUG` guards around the `OPAL_OUTPUT_VERBSE()` call which prints `stop-start`?

File: ompi/communicator/ft/comm_ft.c [hunk 3]
```diff
@@ -210,7 +222,9 @@ int ompi_comm_shrink_internal(ompi_communicator_t* comm, ompi_communicator_t** n
         exit_status = MPI_ERR_INTERN;
         goto cleanup;
     }
+#if OPAL_ENABLE_DEBUG
     stop = PMPI_Wtime();
+#endif
     OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
                          "%s ompi: comm_shrink: GRP COMPUTATION: %g seconds\n",
                          OMPI_NAME_PRINT(OMPI_PROC_MY_NAME), stop-start));
```
Comment (gpaulsen) [inline ompi/communicator/ft/comm_ft.c:230]: I think we also need `#if OPAL_ENABLE_DEBUG` guards around the `OPAL_OUTPUT_VERBSE()` call which prints `stop-start`?

File: ompi/communicator/ft/comm_ft.c [hunk 4]
```diff
@@ -236,7 +252,9 @@ int ompi_comm_shrink_internal(ompi_communicator_t* comm, ompi_communicator_t** n
         exit_status = ret;
         goto cleanup;
     }
+#if OPAL_ENABLE_DEBUG
     stop = PMPI_Wtime();
+#endif
     OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
                          "%s ompi: comm_shrink: NEXT CID: %g seconds\n",
                          OMPI_NAME_PRINT(OMPI_PROC_MY_NAME), stop-start));
```
Comment (gpaulsen) [inline ompi/communicator/ft/comm_ft.c:260]: I think we also need `#if OPAL_ENABLE_DEBUG` guards around the `OPAL_OUTPUT_VERBSE()` call which prints `stop-start`?

File: ompi/communicator/ft/comm_ft.c [hunk 5]
```diff
@@ -261,7 +281,9 @@ int ompi_comm_shrink_internal(ompi_communicator_t* comm, ompi_communicator_t** n
         exit_status = ret;
         goto cleanup;
     }
+#if OPAL_ENABLE_DEBUG
     stop = PMPI_Wtime();
+#endif
     OPAL_OUTPUT_VERBOSE((10, ompi_ftmpi_output_handle,
                          "%s ompi: comm_shrink: COLL SELECT: %g seconds\n",
                          OMPI_NAME_PRINT(OMPI_PROC_MY_NAME), stop-start));
```
Comment (gpaulsen) [inline ompi/communicator/ft/comm_ft.c:289]: I think we also need `#if OPAL_ENABLE_DEBUG` guards around the `OPAL_OUTPUT_VERBSE()` call which prints `stop-start`?

File: ompi/mca/sharedfp/lockedfile/sharedfp_lockedfile_seek.c [hunk 6]
```diff
@@ -119,7 +121,7 @@ mca_sharedfp_lockedfile_seek (ompio_file_t *fh,
 	 *--------------------
 	 */
 	lseek ( fd_lockedfilehandle, 0, SEEK_SET);
-        write ( fd_lockedfilehandle, &offset, sizeof(OMPI_MPI_OFFSET_TYPE));
+        opal_fd_write ( fd_lockedfilehandle, sizeof(OMPI_MPI_OFFSET_TYPE), &offset);
```
Comment (gpaulsen) [inline ompi/mca/sharedfp/lockedfile/sharedfp_lockedfile_seek.c:124]: good catch!

**Review by gpaulsen — CHANGES_REQUESTED** [review]
Just a few more guards needed.

**Comment by gpaulsen** [comment]
Also, strange git-commit-checker error.  Anyone know how to parse that?

```
Run $GITHUB_WORKSPACE/.github/workflows/git-commit-checks.py
Traceback (most recent call last):
  File "/home/runner/work/ompi/ompi/.github/workflows/git-commit-checks.py", line 353, in <module>
    main()
  File "/home/runner/work/ompi/ompi/.github/workflows/git-commit-checks.py", line 337, in main
    check_github_pr_description(config)
  File "/home/runner/work/ompi/ompi/.github/workflows/git-commit-checks.py", line 304, in check_github_pr_description
    if NACP in pr.body:
TypeError: argument of type 'NoneType' is not iterable
Error: Process completed with exit code 1.
```

File: ompi/communicator/ft/comm_ft.c (hunk 2 above)
Comment (bwbarrett) [reply ompi/communicator/ft/comm_ft.c:170]: All caps OPAL_OUTPUT_VERBOSE is the version that only expands to something with --enable-debug.  That's why there was a warning in the first place.

**Comment by jsquyres** [comment]
> Also, strange git-commit-checker error. Anyone know how to parse that?

Looks like it was a transient error; I re-ran it and it was fine.

**Comment by bwbarrett** [comment]
@gpaulsen I think your concerns are wrong; can you have a look at my response and re-review?

File: ompi/communicator/ft/comm_ft.c (hunk 2 above)
Comment (gpaulsen) [reply ompi/communicator/ft/comm_ft.c:170]: AH I see.  So when `OPAL_ENABLE_DEBUG` isn't set, it expands to nothing, meaning no usage of `stop-start`!  Thanks.

**Review by gpaulsen — APPROVED** [review]

**Comment by awlauria** [comment]
Refs: #9097

**Pull request merged**